review/internal/repository/memdb/memdb.go

```diff
@@ -1,34 +1,70 @@
 package memdb
 
 import (
-	"coupon_service/internal/service/entity"
 	"fmt"
-)
-
-type Config struct{}
+	"sync"
 
-type repository interface {
-	FindByCode(string) (*entity.Coupon, error)
-	Save(entity.Coupon) error
-}
+	"coupon_service/internal/service/entity"
+)
 
 type Repository struct {
+	sync.RWMutex
```

Comment:
I made memDB safe for concurrent access using RWMutex, other option is sync.Map.